### PR TITLE
refactor: open file browser and file picker via event emitter

### PR DESCRIPTION
**Pull request by UncleBill**
fixes MF-2971
refine composition dialog, allow to open with initial metas

https://mask.atlassian.net/browse/MF-2971

### CODE REVIEW

File: packages/plugins/FileService/src/SNSAdaptor/FileServiceInjection.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    const [selectMode, setSelectMode] = useState(false)
+    useEffect(() => {
+        const unsubscribe = emitter.on('open', (options) => {
+            setOpen(true)
+            setSelectMode(options.selectMode)
+            setSelectedFiles(options.selectedFiles ?? EMPTY_LIST)
+        })
+        return () => {
+            unsubscribe()
+        }
```
Comment (guanbinrui) [inline packages/plugins/FileService/src/SNSAdaptor/FileServiceInjection.tsx:18]:
```suggestion
        return emitter.on('open', (options) => {
            setOpen(true)
            setSelectMode(options.selectMode)
            setSelectedFiles(options.selectedFiles ?? EMPTY_LIST)
        })
```

Comment (UncleBill) [reply packages/plugins/FileService/src/SNSAdaptor/FileServiceInjection.tsx:18]: `emitter.on` returns `() => boolean`

Comment (guanbinrui) [reply packages/plugins/FileService/src/SNSAdaptor/FileServiceInjection.tsx:18]: Are a cleaner returns boolean not allowed?

**Review by guanbinrui — APPROVED** [review]

File: packages/plugins/FileService/src/SNSAdaptor/FileServiceInjection.tsx (hunk 1 above)
Comment (UncleBill) [reply packages/plugins/FileService/src/SNSAdaptor/FileServiceInjection.tsx:18]: > Are a cleaner returns boolean not allowed?

Yep. Doesn't satisfy typings.

**Pull request merged**